utils/michigan.py

```diff
@@ -0,0 +1,1052 @@
+
+        Inputs:
+            data: a list of 1 or 3 dataframes as outputted from standardize method.
+
+        Returns: (individuals_table, organizations_table, transactions_table)
+                    tuple containing the tables as defined in database schema
+        """
+        individuals_table, individuals_id_mapping = self.create_individuals_table(data)
+        organizations_table, organizations_id_mapping = self.create_organizations_table(
+            data
+        )
+        transactions_table, transactions_id_mapping = self.create_transactions_table(
+            data
+        )
+
+        self.output_id_mapping(
+            individuals_id_mapping, organizations_id_mapping, transactions_id_mapping
+        )
+
+        return (individuals_table, organizations_table, transactions_table)
+
+    # NOTE: The helper functions for ID_mapping output are below
+
+    def output_id_mapping(
+        self,
+        individuals_map: pd.DataFrame,
+        organizations_map: pd.DataFrame,
+        transactions_map: pd.DataFrame,
+    ) -> None:
+        """Creates MichiganIDMAp.csv
+
+        Inputs:
+            individuals_map: dataframe of individuals mapped to database
+            and provided uuid
+            organizations_map: dataframe of organizations mapped to database
+            and provided uuid
+            transactions_map: database of transactions mapped to databasee
+            and provided uuid
+
+        Returns: None, Creates data/output/MichiganIDMAp.csv
+        """
+        output_path = BASE_FILEPATH / "output" / "MichiganIDMap.csv"
+
+        michigan_id_map = pd.concat(
+            [individuals_map, organizations_map, transactions_map], ignore_index=True
+        )
+
+        if not os.path.exists(output_path.parent):
+            os.makedirs(output_path.parent)
+
+        michigan_id_map.to_csv(output_path, index=False)
+
+    def create_individuals_id_mapping(
+        self, individuals: pd.DataFrame, candidates: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Creates the ID mapping dataframe for individuals
+
+        Inputs:
+            individuals: DataFrame with individuals data
+            candidates: DataFrame with candidates data
+
+        Returns: id_mapping: dataframe in the ID mapping format
+        """
+        individuals = individuals[["year", "full_name_uuid"]].copy()
+        candidates = candidates[["year", "candidate_full_name_uuid"]].copy()
+
+        individuals = individuals.rename(columns={"full_name_uuid": "database_id"})
+        candidates = candidates.rename(
+            columns={"candidate_full_name_uuid": "database_id"}
+        )
+
+        id_mapping = pd.concat([individuals, candidates], ignore_index=True)
+        id_mapping["state"] = "MI"
+        id_mapping["entity_type"] = "Individual"
+        id_mapping["provided_id"] = np.nan
+
+        id_mapping = id_mapping[self.id_mapping_column_order].copy()
+
+        return id_mapping
+
+    def create_organizations_id_mapping(
```

Comment:
also unclear why this is split out

Reply:
the id_mapping for each of the tables couldn't be done at once since the year columns were dropped before then per the schema. So I had to add the id mapping into create_tables(). Please let me know if there is a better way to do this

Reply:
It would essentially take the same code that's written above to do that. The whole process would just be moved into the transactions functions. Should i do that before Thursday then?